src/app/pages/vm/vm-wizard/vm-wizard.component.ts

```diff
@@ -285,17 +285,14 @@ export class VmWizardComponent implements OnInit {
   private getGpuRequests(vm: VirtualMachine): Observable<unknown> {
     const gpusIds = this.gpuForm.gpus as unknown as string[];
 
-    const pciIdsRequests$ = gpusIds.map((gpu) => {
-      return this.ws.call('vm.device.get_pci_ids_for_gpu_isolation', [gpu]);
-    });
-
-    return forkJoin(pciIdsRequests$).pipe(
+    return this.ws.call('system.advanced.update_gpu_pci_ids', [gpusIds]).pipe().pipe(
       defaultIfEmpty([]),
-      map((pciIds) => pciIds.flat()),
-      switchMap((pciIds) => forkJoin([
-        this.vmGpuService.updateVmGpus(vm, gpusIds.concat(pciIds)),
-        this.gpuService.addIsolatedGpuPciIds(gpusIds.concat(pciIds)),
-      ])),
+      switchMap(() => {
+        return forkJoin([
+          this.vmGpuService.updateVmGpus(vm, gpusIds),
+          this.gpuService.addIsolatedGpuPciIds(gpusIds),
```

Comment:
So you call `system.advanced.update_gpu_pci_ids` and then you call `addIsolatedGpuPciIds` which also calls the same endpoint.